src/pages/build-apps/examples/heystack.md

```diff
@@ -0,0 +1,482 @@
+---
+title: Heystack app
+description: Interacting with the wallet and smart contracts from a React application
+tags:
+  - example-app
+images:
+  large: /images/pages/heystack-app.svg
+---
+
+## Introduction
+
+This example application demonstrates important features of the Stacks blockchain, and is a case study for how a frontend
+web application can interact with a Clarity smart contract. The full source of the application is provided, this page
+highlights important code snippets and design patterns to help you learn how to develop your own Stacks application.
+
+This app highlights the following platform features:
+
+- Authenticating users with the web wallet
+- Using a smart contract to store data on the blockchain
+- Minting new fungible tokens with a [SIP-010][] compliant smart contract
+- Creating and monitoring transactions on the Stacks blockchain using Stacks.js
+
+You can access the [online version][heystack] of the Heystack app to interact with it. The source for Heystack is also
+available on [Github][heystack_gh]. This page assumes that you are familiar with [React][].
+
+## Heystack overview
+
+Heystack is a web application for chatting with other Stacks users. The application uses the [Stacks web wallet][] to
+authenticate users in the frontend. When a user logs in to Heystack, they're given a genesis amount of the $HEY
+fungible token, which allows them to send and like messages on the platform.
+
+Heystack is powered by Clarity smart contracts so each message is a transaction on the Stacks blockchain. Each time a
+user sends a message on the platform, they must sign the message with the [Stacks web wallet][] (or another compatible
+wallet) and pay a small gas fee in STX. A user spends a $HEY token to send every message, and recieves a $HEY token for
+every like that their messages receive.
+
+## Review smart contracts
+
+Heystack depends on two smart contracts to execute the backend functions of the app on the Stacks blockchain: a contract
+for handling the messaging content, and a contract for minting and distributing the $HEY token.
+
+### Content contract
+
+The `hey.clar` contract provides two primary functions for the application, a function to publish content to
+the blockchain and a function to like a piece of content based on its ID. This section reviews the implementation of
+these primary functions, but is not a comprehensive discussion of the contract.
+
+In order to accomplish the two primary functions, the contract relies on a data variable `content-index` and two
+[data maps][], `like-state` and `publisher-state` which contain the number of likes a piece of content has received, and
```

Comment:
Is it typical for documentation links like this to open in the same tab? I find it to be really confusing in our documentation b/c the `Back` button in the top left doesn't actually go back to where you were. I feel like links should open in a new tab? Not a big deal, but I've definitely gotten lost a few times jumping from link to link.

Reply:
So the way the site is designed is that any internal links will open in the same tab whereas any external links will open in a new tab. That's generally the design I've seen for other documentation sites as well, but I agree our `Back` button causes confusion. It's outside the scope of this PR, but I do plan on revisiting the navigation design of the site at some point in the near future.